### DISCUSSION TITLE
solved

### DISCUSSION THREAD

**Discussion by frazanolo** [tweepy/tweepy#1350] [Questions] [answered]
solved

**Comment by MichaelCurrin** [comment]
Take the streaming filter and run it alone. With a hand coded list of handles.

Perhaps you need @ to be added or removed from handle.

Also BTW you can use Tweepy to lookup a user by name (or ID) and get their ID on the returned User object. You might even be able to do batch lookup like the batch tweet IDs lookup

**Comment by MichaelCurrin** [comment]
Maybe there is just low activity? Try follow your own account and make a tweet.


From tweepy docs:

> This example shows how to use filter to stream tweets by a specific user. The follow parameter is an array of IDs.
> myStream.filter(follow=["2211149702"])

See if that works for our own account and your own stream instance.

See also this issue #981 - the description says there is a limitation. I don't know if that's true.

> `stream.filter(follow=['12345'])` includes tweets at someone and RT

The Twitter docs describe _follow_ here: https://developer.twitter.com/en/docs/tweets/filter-realtime/guides/basic-stream-parameters.

If you don't come right, use `.search` API to get the last 20 or so tweets from named users, though of course you'd have to run that periodically to pull in live tweets.

**Comment by frazanolo** [comment]
i've been doing this with my account since i started and the problem is not that because when i run the single tweepy file it works perfectly, but if i ask it to follow the accounts that i got from the other file it doesn't work anymore

**Comment by Harmon758** [comment, marked as answer]
Since discord.py bots run asynchronously, you'll need to [use the `is_async` parameter to run the stream in a separate thread](https://tweepy.readthedocs.io/en/latest/streaming_how_to.html#async-streaming).
Otherwise, the stream will [block](https://discordpy.readthedocs.io/en/latest/faq.html#what-does-blocking-mean) the Discord bot from running.

Note, Requests is also blocking and is liable to cause issues with your Discord bot.

You can also use Twitter's API with [the GET users/show endpoint](https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/get-users-show) using [`API.get_user`](https://tweepy.readthedocs.io/en/latest/api.html#API.get_user) or [the GET users/lookup endpoint](https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/get-users-lookup) using [`API.lookup_users`](https://tweepy.readthedocs.io/en/latest/api.html#API.lookup_users) to retrieve [User objects](https://developer.twitter.com/en/docs/tweets/data-dictionary/overview/user-object) by ID with attributes including `name`, `screen_name`, etc. Note though, that Tweepy also uses Requests internally. For asyncio support for Tweepy, see #732.
